src/plugins/dashboard/public/dashboard_app/dashboard_app.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { render, waitFor } from '@testing-library/react';
+import { MemoryHistory, createMemoryHistory } from 'history';
+import React, { useEffect } from 'react';
+import type { DashboardRendererProps } from '../dashboard_container/external_api/dashboard_renderer';
+import { buildMockDashboard } from '../mocks';
+import { DashboardApp } from './dashboard_app';
+
+import * as dashboardRendererStuff from '../dashboard_container/external_api/lazy_dashboard_renderer';
+import { DashboardApi } from '..';
+
+/* These tests circumvent the need to test the router and legacy code
+/* the dashboard app will be passed the expanded panel id from the DashboardRouter through mountApp()
+/* @link https://github.com/elastic/kibana/pull/190086/
+*/
+
+describe('Dashboard App', () => {
+  const mockDashboard = buildMockDashboard();
+  let mockHistory: MemoryHistory;
+  let historySpy: jest.SpyInstance;
+  let mockDashboardApiExpandedPanel: string | undefined;
+  const expandPanelSpyInRouter = jest.spyOn(mockDashboard, 'expandPanel');
+
+  beforeEach(() => {
+    mockHistory = createMemoryHistory();
+    historySpy = jest.spyOn(mockHistory, 'replace');
+
+    /**
+     * Mock the LazyDashboardRenderer component to avoid rendering the actual dashboard
+     * and hitting errors that aren't relevant
+     */
+    jest
+      .spyOn(dashboardRendererStuff, 'LazyDashboardRenderer')
+      // we need overwrite the onApiAvailable prop to get the dashboard Api in the dashboard app
+      .mockImplementation(({ onApiAvailable }: DashboardRendererProps) => {
+        useEffect(() => {
+          // mock up similar to the onApiAvailable in DashboardApp
+          if (mockDashboardApiExpandedPanel) {
+            mockDashboard.expandPanel(mockDashboardApiExpandedPanel);
+          }
+          onApiAvailable?.(mockDashboard as DashboardApi);
+        }, [onApiAvailable]);
+
+        return <div>Test renderer</div>;
+      });
```

Comment:
Why is this in a `beforeEach` again? 😅

Reply:
It fails in a beforeAll because of these lines:
``` if (mockDashboardApiExpandedPanel) {
            mockDashboard.expandPanel(mockDashboardApiExpandedPanel);
          }
          onApiAvailable?.(mockDashboard as DashboardApi);
        }, [onApiAvailable]);
```

The mock dashboardRenderer stuff needs to run again for the second test for the expandedPanel condition.

Reply:
Uh oh. You shouldn't be calling `expandPanel` in the spy. Can you explain why you added this?